launchdarkly/resource_environment.go

```diff
@@ -0,0 +1,146 @@
+package launchdarkly
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"sync"
+)
+
+// Since we cannot delete the last environment in a project, we use a hack with a temporary dummy
+// environment that is created when we want to delete the last one. To avoid race conditions when
+// deleting multiple resources (terraform by default runs such operations concurrently), we need
+// to use a mutex.
+var environmentMutex = &sync.Mutex{}
+
+func resourceEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceEnvironmentCreate,
+		Read:   resourceEnvironmentRead,
+		Update: resourceEnvironmentUpdate,
+		Delete: resourceEnvironmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"color": &schema.Schema{
```

Comment:
You could add a ValidateFunc here to verify that this is always an RGB code.

Reply:
The backend returns a good error in that case, but you're right that it'd be better if I add validations at the provider level too. Will do.

Reply:
Here you go.